slosh/tests/lisp-scripts.rs

```diff
@@ -34,48 +36,217 @@ pub fn get_globs_from_test_directory(glob: &str) -> Vec<PathBuf> {
+                        if let Some(file_name) = path.file_name() {
+                            let file_name_str = file_name.to_string_lossy();
+                            if !file_name_str.eq_ignore_ascii_case("README.md") {
+                                results.push(path);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    results
+}
+
+/// Get test files based on command line argument or default behavior
+fn get_test_files() -> Vec<PathBuf> {
+    // Check for command line argument
+    let args: Vec<String> = env::args().collect();
+
+    // With harness = false, arguments come directly to us
+    // Skip the first argument (the program name) and look for a path
+    let target_path = if args.len() > 1 && !args[1].starts_with("--") {
+        Some(&args[1])
+    } else {
+        None
+    };
+
+    if let Some(path_str) = target_path {
+        let path = PathBuf::from(path_str);
+
+        if path.is_file() && path.extension().map(|ext| ext == "slosh").unwrap_or(false) {
+            // Single file specified
+            vec![path]
+        } else if path.is_dir() {
+            // Directory specified - recursively find all .slosh files
+            find_slosh_files_recursive(&path)
+        } else {
+            // Try as a pattern relative to tests directory
+            let tests_dir = get_tests_directory();
+            let full_path = tests_dir.join(&path);
+
+            if full_path.is_file()
+                && full_path
+                    .extension()
+                    .map(|ext| ext == "slosh")
+                    .unwrap_or(false)
+            {
+                vec![full_path]
+            } else if full_path.is_dir() {
+                find_slosh_files_recursive(&full_path)
+            } else {
+                // Fall back to glob pattern
+                get_globs_from_test_directory(path_str)
+            }
+        }
+    } else {
+        // No argument - run all tests recursively
+        find_slosh_files_recursive(&get_tests_directory())
+    }
+}
+
+/// To run all tests in this executable:
+///
+/// ```
+/// # Run all tests
+/// cargo test --package slosh --test lisp-scripts
+///
+/// # Run just string tests
+/// cargo test --package slosh --test lisp-scripts string/
 ///
+/// # Run just math tests
+/// cargo test --package slosh --test lisp-scripts math/
+///
+/// # Run a specific test file
+/// cargo test --package slosh --test lisp-scripts string/str-append.slosh
 /// ```
-/// cargo test --package slosh --test lisp-scripts -- --nocapture
-///```
 ///
-/// All slosh scripts in the ./slosh/tests/ directory will be run by this integration test. If
+/// All slosh scripts found will be run by this integration test. If
 /// the suffix of the file is _fail.slosh, then the test script is expected to fail. All other
 /// slosh scripts should return a 0 exit code or the integration test will fail.
 fn lisp_scripts() {
```

Comment:
the main reason I made this was because I desperately needed shorter feedback loops. I like the tests in the documentation but I don't like not being able to target a given test or set of tests (rn we can only run all of them).

instead of making it s.t. you could tell the test framework to only run specific unit tests, I thought i'd make a more general purpose thing that can execute code in slosh files. esp. the ability to specify a directory and only run things from there.